src/main/interface/msp.c

```diff
@@ -1448,6 +1448,45 @@ static mspResult_e mspFcProcessOutCommandWithArg(uint8_t cmdMSP, sbuf_t *src, sb
             *mspPostProcessFn = mspRebootFn;
         }
 
+        break;
+    case MSP_MULTIPLE_MSP:
+        {
+            #define MAX_MSP_MULTIPLE_REQUESTS 20
+            uint8_t mspCmdBuf[MAX_MSP_MULTIPLE_REQUESTS];
+            uint8_t mspCmdSize[MAX_MSP_MULTIPLE_REQUESTS];
+            uint8_t maxMSPs = 0;
+            if (sbufBytesRemaining(src) == 0) { 
+                return MSP_RESULT_ERROR;
+            }
+            const int maxBytes = sbufBytesRemaining(dst);
+            int bytesRemaining = maxBytes;
+            mspPacket_t packetIn, packetOut;
+            sbufInit(&packetIn.buf, src->ptr, src->end);
+            sbufInit(&packetOut.buf, dst->ptr, dst->end);
+            while (sbufBytesRemaining(src) && maxMSPs < MAX_MSP_MULTIPLE_REQUESTS) {
+                uint8_t newMSP = sbufReadU8(src);
+                mspCmdBuf[maxMSPs] = newMSP;
+                uint8_t* resetPtr = sbufPtr(&packetOut.buf);
+                packetIn.cmd = newMSP;
+                mspFcProcessCommand(&packetIn, &packetOut, mspPostProcessFn);
```

Comment:
Not sure if I understood you correctly, but here is my take on this:
If `mspFcProcessCommand` starts incrementing the pointer in the `packetIn.buf` it won't affect the pointer in `src`, since that is stored in a different struct. I had a dummy uint8_t as the initializer for `sbufInit` for a while but then changed it back to `src` since it doesn't matter what happens to it anyhow.

Reply:
Yes, you are right .. it would require a bit more pointer juggling, not worth it ...